src/librbd/io/ImageRequestWQ.h

```diff
@@ -28,17 +28,21 @@ class ImageRequestWQ : protected ThreadPool::PointerWQ<ImageRequest<ImageCtx> >
   ssize_t read(uint64_t off, uint64_t len, ReadResult &&read_result,
                int op_flags);
   ssize_t write(uint64_t off, uint64_t len, bufferlist &&bl, int op_flags);
-  int discard(uint64_t off, uint64_t len);
+  int discard(uint64_t off, uint64_t len, bool force_discard=false);
```

Comment:
Nit: avoid defaulting the param and instead just fix the compile errors. Especially important given the comment re: changing the logic to just pass "skip partial discard" bool around instead of "force"